content/docs/context.md

```diff
@@ -4,220 +4,110 @@ title: Context
+> Note
+> 
+> For more information about this pattern, see [render props](/docs/render-props.html).
 
-```javascript
-import PropTypes from 'prop-types';
+## Examples
 
-const Button = ({children}, context) =>
-  <button style={{background: context.color}}>
-    {children}
-  </button>;
+### Dynamic Context
 
-Button.contextTypes = {color: PropTypes.string};
-```
+A more complex example with dynamic values for the theme:
 
-## Updating Context
+**theme-context.js**
+`embed:context/theme-detailed-theme-context.js`
 
-Don't do it.
+**themed-button.js**
+`embed:context/theme-detailed-themed-button.js`
 
-React has an API to update context, but it is fundamentally broken and you should not use it.
+**app.js**
+`embed:context/theme-detailed-app.js`
 
-The `getChildContext` function will be called when the state or props changes. In order to update data in the context, trigger a local state update with `this.setState`. This will trigger a new context and changes will be received by the children.
+## Consuming Multiple Contexts
 
-```javascript
-import PropTypes from 'prop-types';
+`embed:context/multiple-contexts.js`
 
-class MediaQuery extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {type:'desktop'};
-  }
+## Accessing Context in Lifecycle Methods
 
-  getChildContext() {
-    return {type: this.state.type};
-  }
+`embed:context/lifecycles.js`
 
-  componentDidMount() {
-    const checkMediaQuery = () => {
-      const type = window.matchMedia("(min-width: 1025px)").matches ? 'desktop' : 'mobile';
-      if (type !== this.state.type) {
-        this.setState({type});
-      }
-    };
+## Forwarding Refs to Context Consumers
 
-    window.addEventListener('resize', checkMediaQuery);
-    checkMediaQuery();
-  }
+**fancy-button.js**
+`embed:context/forwarding-refs-fancy-button.js`
 
-  render() {
-    return this.props.children;
-  }
-}
+**app.js**
+`embed:context/forwarding-refs-app.js`
 
-MediaQuery.childContextTypes = {
-  type: PropTypes.string
-};
-```
+## Legacy API
 
-The problem is, if a context value provided by component changes, descendants that use that value won't update if an intermediate parent returns `false` from `shouldComponentUpdate`. This is totally out of control of the components using context, so there's basically no way to reliably update the context. [This blog post](https://medium.com/@mweststrate/how-to-safely-use-react-context-b7e343eff076) has a good explanation of why this is a problem and how you might get around it.
+> Note
+> 
+> React previously shipped with an experimental context API. The old API will be supported in all 16.x releases, but applications using it should migrate to the new version. The legacy API will be removed in React 17. Read the [legacy context docs here](/docs/legacy-context.html).
```

Comment:
"in a future major React version", let's not be explicit here (we aren't actually sure yet)